### PR TITLE
Implement keyboard shortcuts for Voice assignment

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: https://github.com/musescore/MuseScore/issues/23370

### CODE REVIEW

File: src/notation/internal/notationactioncontroller.h [hunk 1]
```diff
@@ -247,8 +247,8 @@ class NotationActionController : public muse::actions::Actionable, public muse::
     template<typename P1>
     void registerAction(const muse::actions::ActionCode&, void (INotationInteraction::*)(P1), P1,
                         bool (NotationActionController::*)() const);
-    template<typename P1, typename P2>
-    void registerAction(const muse::actions::ActionCode&, void (INotationInteraction::*)(P1, P2), P1, P2, PlayMode = PlayMode::NoPlay,
+    template<typename P1, typename P2, typename Q1, typename Q2>
+    void registerAction(const muse::actions::ActionCode&, void (INotationInteraction::*)(P1, P2), Q1, Q2, PlayMode = PlayMode::NoPlay,
```
Comment (cbjeukendrup) [inline src/notation/internal/notationactioncontroller.h:251]: This change is necessary to be able to pass `int` literals to `size_t` arguments. Under normal circumstances, that just works, but here with this template stuff, the compiler behaves very pedantic and a bit dumb. If the given function takes `size_t` but the given parameters are `int`, then it says "I can't deduce what P1 and P2 should be, namely either `size_t` or `int`". Normally, the parameters would implicitly be converted to `size_t`, and that is emulated by this change.

If this change is undesired, I can just drop that second commit.

**Review by RomanPudashkin — APPROVED** [review]

**Review by avvvvve — APPROVED** [review]
superb!

**Comment by miiizen** [comment]
Also resolves: https://github.com/musescore/MuseScore/issues/23543!

**Comment by cbjeukendrup** [comment]
Ah yes, I didn't remember that that issue existed :)

**Pull request merged**